homeassistant/helpers/template.py

```diff
@@ -1352,6 +1360,32 @@ def __init__(self, hass):
         self.globals["strptime"] = strptime
         self.globals["urlencode"] = urlencode
         if hass is None:
+
+            def unsupported(name):
+                def warn_unsupported(*args, **kwargs):
+                    _LOGGER.warning(
+                        "Use of '%s' is not supported in limited templates", name
+                    )
+                    return ""
```

Comment:
Should it raise?

Reply:
Not sure, maybe better to raise to entirely block the template from rendering?

Speaking of raising, should we overload the default jinja behavior on undefined variables to raise and/or log a warning instead of silently returning the empty string?
(A general question, not related to this PR)

Reply:
I think it's better to break it and not have it attach the trigger.

Reply:
Because triggers are a pain to debug.

Reply:
Changed to raise, it gets really noisy though:
```
2021-01-27 20:52:58 ERROR (MainThread) [homeassistant.components.automation.hasp_control_slider_object] Error setting up trigger HASP Control Slider Object
Traceback (most recent call last):
  File "/mnt/d/development/github/home-assistant_fork/homeassistant/helpers/template.py", line 357, in async_render
    render_result = compiled.render(kwargs)
  File "/mnt/d/development/github/home-assistant_fork/venv/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/mnt/d/development/github/home-assistant_fork/venv/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/mnt/d/development/github/home-assistant_fork/venv/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/mnt/d/development/github/home-assistant_fork/venv/lib/python3.8/site-packages/jinja2/sandbox.py", line 462, in call
    return __context.call(__obj, *args, **kwargs)
  File "/mnt/d/development/github/home-assistant_fork/homeassistant/helpers/template.py", line 1366, in warn_unsupported
    raise TemplateError(f"Use of '{name}' is not supported in limited templates")
homeassistant.exceptions.TemplateError: str: Use of 'states' is not supported in limited templates

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/d/development/github/home-assistant_fork/homeassistant/components/mqtt/trigger.py", line 47, in async_attach_trigger
    topic = topic.async_render(variables, limited=True)
  File "/mnt/d/development/github/home-assistant_fork/homeassistant/helpers/template.py", line 359, in async_render
    raise TemplateError(err) from err
homeassistant.exceptions.TemplateError: TemplateError: str: Use of 'states' is not supported in limited templates
```

Reply:
We should catch `HomeAssistantError` and then not print the stack trace. That will help :)

Reply:
Yeah, fixed in eeac975